src/app/backend/resource/pod/poddetail.go

```diff
@@ -19,25 +19,57 @@ import (
 	"log"
 
 	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/apps"
 	"k8s.io/kubernetes/pkg/apis/batch"
+	"k8s.io/kubernetes/pkg/apis/extensions"
 	k8sClient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/client/typed/discovery"
 
 	"github.com/kubernetes/dashboard/src/app/backend/client"
 	"github.com/kubernetes/dashboard/src/app/backend/resource/common"
 	"github.com/kubernetes/dashboard/src/app/backend/resource/dataselect"
 	"github.com/kubernetes/dashboard/src/app/backend/resource/job/joblist"
-//	"github.com/kubernetes/dashboard/src/app/backend/resource/replicaset/replicasetlist"
-//	"github.com/kubernetes/dashboard/src/app/backend/resource/replicationcontroller/replicationcontrollerlist"
-//	"github.com/kubernetes/dashboard/src/app/backend/resource/daemonset/daemonsetlist"
-//	"github.com/kubernetes/dashboard/src/app/backend/resource/petset/petsetlist"
+	"github.com/kubernetes/dashboard/src/app/backend/resource/replicaset/replicasetlist"
+	"github.com/kubernetes/dashboard/src/app/backend/resource/replicationcontroller/replicationcontrollerlist"
+	"github.com/kubernetes/dashboard/src/app/backend/resource/daemonset/daemonsetlist"
+	"github.com/kubernetes/dashboard/src/app/backend/resource/petset/petsetlist"
 	"github.com/kubernetes/dashboard/src/app/backend/resource/metric"
 )
 
-// TODO: use a centralized place to store this constant
+// TODO: Update vendor/k8s.io
 const (
 	CreatedByAnnotation = "kubernetes.io/created-by"
 )
 
+// Interface holds the methods for clients of Kubernetes,
+// an interface to allow mock testing.
+type Interface interface {
+	k8sClient.PodsNamespacer
+	k8sClient.PodTemplatesNamespacer
+	k8sClient.ReplicationControllersNamespacer
+	k8sClient.ServicesNamespacer
+	k8sClient.EndpointsNamespacer
+	k8sClient.NodesInterface
+	k8sClient.EventNamespacer
+	k8sClient.LimitRangesNamespacer
+	k8sClient.ResourceQuotasNamespacer
+	k8sClient.ServiceAccountsNamespacer
+	k8sClient.SecretsNamespacer
+	k8sClient.NamespacesInterface
+	k8sClient.PersistentVolumesInterface
+	k8sClient.PersistentVolumeClaimsNamespacer
+	k8sClient.ComponentStatusesInterface
+	k8sClient.ConfigMapsNamespacer
+	Apps() k8sClient.AppsInterface //Added this one
+	Autoscaling() k8sClient.AutoscalingInterface
+	Batch() k8sClient.BatchInterface
+	Extensions() k8sClient.ExtensionsInterface
+	Rbac() k8sClient.RbacInterface
+	Discovery() discovery.DiscoveryInterface
+	Certificates() k8sClient.CertificatesInterface
+}
+// End TODO:
```

Comment:
We really need to update the vendor/k8s.io code now to get PetSets all around in the api. Embedding this interface here makes me trick the go type system for a bit, but this won't help for the fakeClient used by testing